src/game.cpp

```diff
@@ -10351,6 +10351,15 @@ int game::move_liquid(item &liquid)
 
 void game::drop(int pos)
 {
+    if (!m.can_put_items(u.pos())) {
+        int part = -1;
+        vehicle * const veh = m.veh_at( u.pos(), part );
+        if( veh == nullptr || veh->part_with_feature( part, "CARGO" ) < 0 ) {
+            add_msg(m_info, _("You can't place items there!"));
+            return;
+        }
+    }
+
```

Comment:
I'd suggest extracting a function which could be invoked from both `drop` and `drop_in_direction`.